public/controllers/register-agent/services/register-agent-os-commands-services.tsx

```diff
@@ -0,0 +1,109 @@
+import { tOptionalParameters } from '../core/config/os-commands-definitions';
+import {
+  IOptionalParameters,
+  tOSEntryInstallCommand,
+  tOSEntryProps,
+} from '../core/register-commands/types';
+import { tOperatingSystem } from '../hooks/use-register-agent-commands.test';
+
+const getAllOptionals = (
+  optionals: IOptionalParameters<tOptionalParameters>,
+  osName?: tOperatingSystem['name'],
+) => {
+  // create paramNameOrderList, which is an array of the keys of optionals add interface
+  const paramNameOrderList: (keyof IOptionalParameters<tOptionalParameters>)[] =
+    ['serverAddress', 'wazuhPassword', 'agentGroups', 'agentName', 'protocol'];
+
+  if (!optionals) return '';
+  let paramsText = Object.entries(paramNameOrderList).reduce(
+    (acc, [key, value]) => {
+      if (optionals[value]) {
+        acc += `${optionals[value]} `;
+      }
+      return acc;
+    },
+    '',
+  );
+
+  if (osName && osName.toLowerCase() === 'windows' && optionals.serverAddress) {
+    // when os is windows we must to add wazuh registration server with server address
+    paramsText =
+      paramsText + `WAZUH_REGISTRATION_SERVER=${optionals.serverAddress.replace('WAZUH_MANAGER=','')} `;
```

Comment:
**nitpick**: replacing the substring could cause problems if there is another unexpected match, we should avoid replacing the string.

Reply:
Yes, this isn't the best solution but this is the only way right now without refactoring the optional return value implementation. To improve this we need to change the way that is returned the params because right now is returned 'PARAM_NAME=VALUE' and we cannot change it depending on the OS selected. Maybe in a next version, we can change it.